### PR TITLE
Instrumentation for Kotlin coroutines 1.5.0+

### PR DESCRIPTION
**Pull request by monosoul**
# What Does This Do
Introduces support for coroutines 1.5.0+. 
Solves #931 , Solves #1123 .

# Motivation
The instrumentation I introduced in #4405 only supports Kotlin coroutines up to 1.4.3. 

# Additional Notes
Updates project-wide Kotlin dependency version to 1.6.21 because latest versions of coroutines require higher API level version (unfortunately Kotlin plugin for Gradle has some breaking changes in version 1.7.0, so I didn't use it here).
Introduces 3 new Gradle modules: `kotlin-coroutines-common`, `kotlin-coroutines-1.3` and `kotlin-coroutines-1.5`.
 Also configures `lastDep` test (was missing before).

### CODE REVIEW

**Comment by monosoul** [comment]
@bantonsson would you mind to have a look since you reviewed the last one? Thanks!

**Comment by monosoul** [comment]
Hey @bantonsson , sorry for pinging, but I'd appreciate it a lot if we can get this merged before the next release.
I even got lucky enough to get the `required` build green this time :grin:

File: dd-java-agent/instrumentation/kotlin-coroutines-1.5/src/test/groovy/KotlinCoroutineInstrumentationTest.groovy [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import kotlinx.coroutines.CoroutineDispatcher
+
+class KotlinCoroutineInstrumentationTest extends AbstractKotlinCoroutineInstrumentationTest<KotlinCoroutineTests> {
+
+  @Override
+  protected KotlinCoroutineTests getCoreKotlinCoroutineTestsInstance(CoroutineDispatcher dispatcher) {
+    return new KotlinCoroutineTests(dispatcher)
+  }
+
+  def "kotlin traced across channels"() {
```
Comment (bantonsson) [inline dd-java-agent/instrumentation/kotlin-coroutines-1.5/src/test/groovy/KotlinCoroutineInstrumentationTest.groovy:12]: Should probably be `"kotlin traced across flows"`

**Review by bantonsson — APPROVED** [review]
Thanks a lot @monosoul. Would you mind squashing this as well?

Only minor nitpick. Going forward we want to try to group related instrumentations in this fashion:
```
:dd-java-agent:instrumentation:kotlin-coroutines
:dd-java-agent:instrumentation:kotlin-coroutines:coroutines-1.3
:dd-java-agent:instrumentation:kotlin-coroutines:coroutines-1.5
```

File: dd-java-agent/instrumentation/kotlin-coroutines-1.5/src/test/groovy/KotlinCoroutineInstrumentationTest.groovy (hunk 1 above)
Comment (monosoul) [reply dd-java-agent/instrumentation/kotlin-coroutines-1.5/src/test/groovy/KotlinCoroutineInstrumentationTest.groovy]: Thanks for noticing! Fixed here: 24899516a0626fb5cd722eda298eb8ceebab6f4c

**Comment by monosoul** [comment]
> Thanks a lot @monosoul. Would you mind squashing this as well?
> 
> Only minor nitpick. Going forward we want to try to group related instrumentations in this fashion:
> 
> ```
> :dd-java-agent:instrumentation:kotlin-coroutines
> :dd-java-agent:instrumentation:kotlin-coroutines:coroutines-1.3
> :dd-java-agent:instrumentation:kotlin-coroutines:coroutines-1.5
> ```

@bantonsson squashed and refactored the submodules structure :+1:

**Review by bantonsson — APPROVED** [review]

**Pull request merged**